extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -413,20 +413,30 @@ export class AmpAnalytics extends AMP.BaseElement {
    * @param {string} template The template to expand.
    * @param {!JSONObject} The object to use for variable value lookups.
    * @param {!Object} event Object with details about the event.
+   * @param {number} opt_iterations Number of recursive expansions to perform.
+   *    Defaults to 2 substitutions.
    * @return {string} The expanded string.
    * @private
    */
-  expandTemplate_(template, trigger, event) {
+  expandTemplate_(template, trigger, event, opt_iterations) {
+    opt_iterations = opt_iterations === undefined ? 2 : opt_iterations;
+    if (opt_iterations < 0) {
+      return '';
```

Comment:
I did that in the first version. I think it is logical to return template but the recusion messes up the values during unwinding. 

See the updated tests. One thing I could do is to replace just the vars templates with ''. Any suggestions?